### PR TITLE
implemented g-drive data fetching

### PR DESCRIPTION
**Pull request by techsavvyash**

### CODE REVIEW

File: src/db.py [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+
+# queries to create the db tables
+
+
 @dataclass
 class File:
     name: str
     link: str
-    tags: list[str]
+    tags: "list[str]"
```
Comment (maan2003) [inline src/db.py:13]: why?

File: src/db.py [hunk 2]
```diff
@@ -31,9 +64,18 @@ def list_files(query: str, tags: list[str]) -> list[File]:
...
     # TODO: add to db
+    """
+    for file in files:
+        query = f"INSERT INTO files(name, drive_url) VALUES ('{file.name}', '{file.link}');"
+        cursor.execute(query)
+        db.commit()
+        print("added", file.name)
+
+    """
+    print("adding file")
```
Comment (maan2003) [inline src/db.py:80]: remove the changes in `db.py`, these should be in a separate pr

File: src/drive.py [hunk 3]
```diff
@@ -1,10 +1,79 @@
+# import db
+from Google import connect_to_drive
+import pandas as pd
 import db
 
+
+def get_folder_contents(service, id):
+
+    query = f"parents = '{id}'"
```
Comment (maan2003) [inline src/drive.py:9]: this just work for some reason.
```suggestion
    query = f"'{id}' in parents"
```

File: src/drive.py [hunk 4]
```diff
@@ -1,10 +1,79 @@
...
+            except Exception as error:
+                print("Error: ", error)
+
+
 def main():
-    # TODO: query the drive and add files to the database
-    # call db.add_files() to add files to the database
+
+    folder_id = '1U2taK5kEhOiUJi70ZkU2aBWY83uVuMmD'
+    CLIENT_SECRET_FILE = '../credentials.json'
```
Comment (maan2003) [inline src/drive.py:58]: lets keep the file path in repo root. ideally this should be taken as argument to the program
```suggestion
    CLIENT_SECRET_FILE = './credentials.json'
```

File: src/drive.py [hunk 5]
```diff
@@ -1,10 +1,79 @@
...
+def list_drive(service, folder_id, result, parentName='', grandParentName=''):
+    files = get_folder_contents(service, folder_id)
+
+    for item in files:
+        if item['mimeType'] != 'application/vnd.google-apps.folder':
+            try:
+                id = item['id']
+                link = (service.files().get(fileId=id, fields='webViewLink').execute())[
+                    'webViewLink']
+                item['tags'] = [parentName, grandParentName]
```
Comment (maan2003) [inline src/drive.py:35]: we need more than 2 ancestors in many cases

File: src/drive.py [hunk 6]
```diff
@@ -1,10 +1,79 @@
...
+
+
+def list_drive(service, folder_id, result, parentName='', grandParentName=''):
+    files = get_folder_contents(service, folder_id)
+
+    for item in files:
+        if item['mimeType'] != 'application/vnd.google-apps.folder':
+            try:
+                id = item['id']
+                link = (service.files().get(fileId=id, fields='webViewLink').execute())[
```
Comment (maan2003) [inline src/drive.py:33]: we don't need link, id is sufficient

**Review by maan2003 — CHANGES_REQUESTED** [review]

File: src/db.py (hunk 1 above)
Comment (techsavvyash) [reply src/db.py:13]: The linter was showing an error that's why!

Comment (maan2003) [reply src/db.py:13]: which linter?

Comment (techsavvyash) [reply src/db.py:13]: PyFlakes I guess

File: src/db.py (hunk 2 above)
Comment (techsavvyash) [reply src/db.py:80]: removed

File: src/drive.py (hunk 3 above)
Comment (techsavvyash) [reply src/drive.py:9]: updated

File: src/drive.py (hunk 4 above)
Comment (techsavvyash) [reply src/drive.py:58]: updated

File: src/drive.py (hunk 5 above)
Comment (techsavvyash) [reply src/drive.py:35]: did nothing

File: src/drive.py (hunk 6 above)
Comment (techsavvyash) [reply src/drive.py:33]: removed the webViewLink field
